include/SDL3/SDL_filesystem.h

```diff
@@ -236,6 +236,125 @@ typedef enum
+     * \return a RWops for the path specified.
+     */
+    SDL_RWops *(SDLCALL * open)(SDL_FSops *fs, const char *path, const char *mode);
+
+    /* list all files in a directory. Each file is passed through a callback until we're done or the callback returns <= 0. */
+    int (SDLCALL * enumerate)(SDL_FSops *fs, const char *path, SDL_EnumerateCallback cb, void *userdata);
+
+    /* delete/unlink/rmdir a path. Will not remove non-empty directories! */
+    int (SDLCALL * remove)(SDL_FSops *fs, const char *path);
+
+    /* Create a directory. */
+    int (SDLCALL * mkdir)(SDL_FSops *fs, const char *path);
+
+    /* Obtain basic path metadata. */
+    int (SDLCALL * stat)(SDL_FSops *fs, const char *path, SDL_Stat *stat);
+
+    /* Close the filesystem and free its resources. */
+    void (SDLCALL * closefs)(SDL_FSops *fs);
+
+    /* Used by the SDL_FSops for private data. Don't touch! */
+    void *opaque;
+
+    /* Everything has properties! :) */
+    SDL_PropertiesID props;
+};
+
+
+/*
+ * Create an SDL_FSops that represents the native filesystem, treating `basedir` as
+ * the root of the tree. A NULL basedir is valid and gives you the actual root of the
+ * native filesystem.
+ *
+ * Enumerating root on Windows will give you available drive letters like they were directories,
+ * almost everything else assumes the root is a Unix "/" thing.
+ *
+ * Destroy this when done with SDL_DestroyFilesystem().
+ */
+extern DECLSPEC SDL_FSops * SDLCALL SDL_CreateFilesystem(const char *basedir);
+
+
+/* Use the filesystem... */
+extern DECLSPEC SDL_RWops * SDLCALL SDL_FSopen(SDL_FSops *fs, const char *path, const char *mode);
+extern DECLSPEC int SDLCALL SDL_FSenumerate(SDL_FSops *fs, const char *path, SDL_EnumerateCallback cb, void *userdata);
+extern DECLSPEC int SDLCALL SDL_FSremove(SDL_FSops *fs, const char *path);
+extern DECLSPEC int SDLCALL SDL_FSmkdir(SDL_FSops *fs, const char *path);
+extern DECLSPEC int SDLCALL SDL_FSstat(SDL_FSops *fs, const char *path, SDL_Stat *stat);
+
+/* returns `props` field of `context`, creating an SDL_PropertiesID for that field, if it doesn't exist yet. */
+extern DECLSPEC SDL_PropertiesID SDLCALL SDL_GetFSProperties(SDL_FSops *context);
+
+/**
+ * if fs is not NULL, this will call fs->closefs() (if closefs is not NULL), destroy fs->props, and then SDL_free(fs). You can use this
+ * with your own SDL_FSops implementations if that sequence of operations is correct.
+ */
+extern DECLSPEC void SDLCALL SDL_DestroyFilesystem(SDL_FSops *fs);
+
+
+/* some helper functions ... */
+
+/* returns "\\" or "/" etc. Read-only static data, don't free or modify! */
+extern DECLSPEC const char * SDLCALL SDL_GetFilePathSeparator(void);
+
+/* converts an SDL file timestamp into a win32 FILETIME (100-nanosecond intervals since January 1, 1601). Fills in the two 32-bit values of the win32 struct. */
+extern DECLSPEC void SDLCALL SDL_FileTimeToWindows(Sint64 ftime, Uint32 *low, Uint32 *high);
```

Comment:
A [`LARGE_INTEFER`](https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-ularge_integer-r1) also has a 64-bit `QuadPart` member.
Perhaps this function can return a Sint64 instead?

Or is another function the troublemaker?

Reply:
QuadPart is also unsigned. We can cast it, but the implication is Windows will never give you a negative FILETIME, so we shouldn't offer it either.